core/src/main/java/contacts/core/Query.kt

```diff
@@ -582,7 +592,7 @@ private fun ContentResolver.resolve(
     // Get Contact Ids matching where from the Data table. If where is null, skip.
     if (where != null && !cancel()) {
         contactIds = mutableSetOf<Long>().apply {
-            addAll(findContactIdsInDataTable(where, cancel))
+            addAll(findContactIdsInDataTable(reduce(where, cancel), cancel))
```

Comment:
Reduce the `where` such that one side of a "where AND where" will have zero mime type fields.